### PR TITLE
fix(geo):  use `label.formatter` instead of ` label.normal.formatter`

### LINKED ISSUE #10191: map series 使用 geoIndex 指定 geo 组件时，geo 的 label.formatter 属性失效
### Version
4.2.1

### Reproduction link
[https://www.echartsjs.com/gallery/editor.html?c=geo-map-scatter](https://www.echartsjs.com/gallery/editor.html?c=geo-map-scatter)

### Steps to reproduce
如[文档](https://www.echartsjs.com/option.html#geo.label.formatter)所示，geo.label.formatter 中添加 formatter 属性，此属性无效。

### What is expected?
地图每一块的标签值随 formatter 的更改而更改。

### What is actually happening?
标签值依旧是 series[i]-map.data[i].name 不变。

---
### 补充

[文档 series[i]-map.geoIndex](https://www.echartsjs.com/option.html#series-map.geoIndex)中说：
> 当设定了 geoIndex 后，series-map.map 属性，以及 series-map.itemStyle 等样式配置不再起作用，而是采用 geo 中的相应属性。

### PR DESCRIPTION
**Pull request by Ende93**
fix #10191

### CODE REVIEW

File: src/coord/geo/GeoModel.js [hunk 1]
```diff
@@ -148,7 +148,7 @@ var GeoModel = ComponentModel.extend({
      */
     getFormattedLabel: function (name, status) {
         var regionModel = this.getRegionModel(name);
-        var formatter = regionModel.get('label.' + status + '.formatter');
+        var formatter = regionModel.get(['label', status === 'normal' ? '' : status, 'formatter'].join('.'));
```
Comment (100pah) [inline src/coord/geo/GeoModel.js]: I think this fix is almost correct.
But there is only a problem that 
if the `status` is 'normal',
`['label', status === 'normal' ? '' : status, 'formatter'].join('.')` will get `'label..formatter'`.

Comment (Ende93) [reply src/coord/geo/GeoModel.js]: fixed and I also fix the formatter params & str replacement, please review it

**Comment by Ende93** [comment]
demo is [here](https://ende93.github.io/echarts/test/geo-map.html), see console and hover map can see the effect

**Comment by shenshuai89** [comment]
@Ende93 I use “npm install echarts --save” run this change，then happen  an error “_Error in mounted hook: "TypeError: mapSeries.getDataParams is not a function"_”

**Comment by Ende93** [comment]
@shenshuai89 can you provide demo ?

**Comment by shenshuai89** [comment]
> @shenshuai89 can you provide demo ?

Demo url  http://lovedesign.com.cn/sam/echart/

**Comment by Ende93** [comment]
@shenshuai89 your echarts's version seem like 4.1.0 which dont have getDataParams function, please try 4.2.1.

**Comment by shenshuai89** [comment]
there are still the same errors

**Comment by Ende93** [comment]
@shenshuai89 hmm, can you give a readable repo? so I can debug it

**Comment by shenshuai89** [comment]
https://github.com/shenshuai89/echartDemo
this is project。

**Comment by Ende93** [comment]
> https://github.com/shenshuai89/echartDemo
> this is project。

Sorry, seem you use 4.2.1 echarts that published, so I still cant see what happend. Can you use my branch code to do the test or show full code?

**Comment by shenshuai89** [comment]
thanks，now have not errors。but can‘t see console.log(params), i dont know the reason

**Pull request closed without merging**